### PR TITLE
Add redux files for actions reducer and store. Add logic for grabbing the FHIR server in context.

### PR DESCRIPTION
**Pull request by zplata**
This PR sets up the necessary files to setup Redux (for data storing) within the application. Additionally, this includes the logic that calls the default/specified FHIR server in context for the application and [stores relevant metadata](https://github.com/cds-hooks/sandbox/blob/de0892339ad56fefa248a172da27f62311c4f3dd/src/reducers/fhir-server-reducers.js). Tests accompany each piece of the logic included for this workflow, excluding the UI component tests for now.

To run: clone the project down and run `npm install` and then `npm run start` and navigate to localhost:8080. 

@kpshek 
@mjhenkes
@tbiethman
@kolkheang

### CODE REVIEW

File: src/reducers/fhir-server-reducers.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import * as types from '../actions/action-types';
+
+const initialState = {
```
Comment (kolkheang) [inline src/reducers/fhir-server-reducers.js:3]: Future: Let's pull these items to a config file.

**Review by kolkheang — APPROVED** [review]

File: src/reducers/fhir-server-reducers.js (hunk 1 above)
Comment (zplata) [reply src/reducers/fhir-server-reducers.js:3]: Each reducer will have it's own part of the application state. So for all fhir-server related attributes we want to store on the entire application state, here is where we define these properties. As this app grows larger later on, we could investigate moving all initial state properties into one file, or config file as you put it, and importing here accordingly.

File: src/actions/fhir-server-actions.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    baseUrl,
+    metadata,
+  };
+}
+
+/**
+ * Signals failed connection to FHIR server in context.
+ * @returns {{type}} - Action to dispatch
+ */
+export function signalFailureFhirServerRetrieval() {
```
Comment (shriniketsarkar-zz) [inline src/actions/fhir-server-actions.js:21]: Do we want to store the error message returned by the service for reporting to the user in some way?

Comment (zplata) [reply src/actions/fhir-server-actions.js:21]: Yeah, it might be in a separate task for error-handling with modals later on.

**Review by shriniketsarkar-zz — APPROVED** [review]

**Review by rishabh92 — APPROVED** [review]

**Review by kpshek — APPROVED** [review]

**Pull request merged**